airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/api.py

```diff
@@ -87,34 +87,41 @@ def _parse_call_rate_header(headers):
 
         return usage, pause_interval
 
-    def handle_call_rate_limit(self, response, params):
+    def _compute_pause_interval(self, usage, pause_interval):
+        """The sleep time will be calculated based on usage consumed."""
+        if usage >= self.MAX_RATE:
+            return max(self.MAX_PAUSE_INTERVAL, pause_interval)
+        return self.MIN_PAUSE_INTERVAL
```

Comment:
```suggestion
        return max(self.MIN_PAUSE_INTERVAL, pause_interval)
```

If the usage is lower than the max rate but the pause interval is higher than the default minimum one: don't we want to be conservative and pause for the time the API requests?

Reply:
> If the usage is lower than the max rate but the pause interval is higher than the default minimum one: don't we want to be conservative and pause for the time the API requests?

for sure, the idea is to pause the requests taking into account the time interval that the response returns.